daffodil-core/src/main/scala/org/apache/daffodil/dsom/InitiatedTerminatedMixin.scala

```diff
@@ -61,7 +64,23 @@ trait InitiatedTerminatedMixin
     res
   }
 
-  private lazy val initiatorDiscriminator = prod("initiatorDiscriminator", parentSaysInitiatedContent) { InitiatedContent(this) }
+  private lazy val initiatorDiscriminator = prod(
+    "initiatorDiscriminator",
+    parentSaysInitiatedContent &&
+      immediatelyEnclosingModelGroup.map {
+        case c: ChoiceTermBase => true
+        case s: SequenceTermBase => (isArray || isOptional) &&
+          isVariableOccurrences
+      }.getOrElse(false)) {
+      this match {
+        case eb: ElementBase => {
+          if (eb.minOccurs < 1)
+            SDW(WarnID.InitiatedContentMinOccursZero, "The minOccurs attribute should not be zero when dfdl:initiatedContent is 'yes'.")
```

Comment:
Should this be an SDE?

Reply:
Yes. I'll add comments. I made this an SDW because making it an SDE breaks EDIFACT, possibly other schemas.
I will add a JIRA ticket to add tunable for this to control SDE/SDW.

Reply:
Actually. This is a bug. The restriction that the child of the initiatedContent group cannot be optional applies only to choices, not to sequences. The EDIFACT situation is sequences. So I can fix this.